### PR TITLE
Add a GrayscaleToRgb transform that can expand channels to 3

### PR DESCRIPTION
**Pull request by ahmadsharif1**
This is for issue #8186.

GrayscaleToRgb transform can take a grayscale image and expand it to 3 channels, with the same values for R, G and B.

### CODE REVIEW

File: torchvision/transforms/v2/functional/_color.py
Comment (NicolasHug) [inline torchvision/transforms/v2/functional/_color.py:89]: Calling `*to_grayscale` will properly make the image 3D, but it will also convert it to grayscale which is not desirable (for both tensors and PIL images). Take a look at the output of this

```py
# %%
import torch
from torchvision.transforms.v2.functional import grayscale_to_rgb, to_pil_image

#%%
img = torch.rand(3, 300, 400)  # RGB
pil_img = to_pil_image(img)  # RGB

# %%
pil_img
# %%
to_pil_image(grayscale_to_rgb(img))  # this is grayscale

# %%

grayscale_to_rgb(to_pil_image(img))  # This is grayscale
```

(open it in vscode to make it a notebook) and you should see that the images are converted to grayscale.

File: torchvision/transforms/transforms.py
Comment (NicolasHug) [inline torchvision/transforms/transforms.py:1580]: Sorry I should have made this clearer earlier, but we shouldn't be changing anything outside of `torchvision.transforms.v2`.

Now that V2 is fully stable, we are only making improvements and changes to the v2 namespace. We're keeping the "v1" namespace around (i.e. `torchvision/transforms/transforms.py`, `torchvision/transforms/functional.py`, `torchvision/transforms/_functional_tensor.py`, `torchvision/transforms/_functional_pil.py`), and we're fixing bug etc., but we don't need to add new features there. So for this PR, we can focus on `torchvision.transforms.v2` only

File: torchvision/transforms/v2/functional/_color.py
Comment (ahmadsharif1) [reply torchvision/transforms/v2/functional/_color.py]: Good catch. I addressed this issue for Tensors and for PILs, now I am simply calling the convert function which deals with this by itself. Also added a test for it.

File: torchvision/transforms/transforms.py
Comment (ahmadsharif1) [reply torchvision/transforms/transforms.py:1580]: I removed this line from the `GrayscaleToRgb` class: `    _v1_transform_cls = _transforms.Grayscale` and reverted the changes in the v1 files. I hope that is sufficient to address this comment.

File: test/test_transforms_v2.py
Comment (NicolasHug) [inline test/test_transforms_v2.py:5028]: You might be able to just get away with this? I'm not sure we need to manually override the values and check again just to make sure that the image is passed-through


```suggestion
        image = make_image(dtype=torch.uint8, device="cpu", color_space="RGB")
        assert_equal(F.grayscale_to_rgb(image), image)
```

**Review by NicolasHug — APPROVED** [review]
Thanks a lot @ahmadsharif1 , just one minor comment below but this looks good. Maybe you can undraft the PR?
Now it's time to bikeshed about the name. I'm tempted to just call that `ToRGB()` instead of `GrayScaleToRGB()` so that it can be called at the beginning of every pipeline without suggesting that the input *has* to be grayscale...

**Comment by NicolasHug** [comment]
As discussed offline, we should just try to be consistent with the existing transforms here (even though the existing transforms themselves are not completely consistent...).

- We already have the `Grayscale` transform, so I guess that means we should just have `RGB()`
- We have the `rgb_to_grayscale()` functional, so lets' have `grayscale_to_rgb()` as well. (We do have `to_grayscale()`, but it's an oddity that only supports PIL images for whatever reason... so we shouldn't build on it)

Also before merging we'll have to add the transform and the functional to the docs [here](https://github.com/pytorch/vision/blob/53869eb8cdc12ebedb020bf93c6bc0cc684bc7b2/docs/source/transforms.rst?plain=1#L349) and [here](https://github.com/pytorch/vision/blob/53869eb8cdc12ebedb020bf93c6bc0cc684bc7b2/docs/source/transforms.rst?plain=1#L367). To check for docs rendering you can either check out instructions in the contributing guide, or (probably easier for now) just wait for the CI job to be done and then check from https://github.com/pytorch/vision/pull/8247#issuecomment-1922504673

File: torchvision/transforms/v2/_color.py
Comment (NicolasHug) [inline torchvision/transforms/v2/_color.py:58]: Sorry, forgot that one (a Python-wide convention)

https://peps.python.org/pep-0257/#one-line-docstrings

> The docstring is a phrase ending in a period. It prescribes the function or method’s effect as a command (“Do this”, “Return that”), not as a description; e.g. don’t write “Returns the pathname …”.

```suggestion
    """Convert images or videos to RGB (if they are already not RGB)
```

**Pull request merged**
